packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -772,6 +791,16 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
                 clientId: this.connectionManager.clientId,
             });
         }
+
+        // Client ops: MSN has to be lower than sequence #, as client can continue to send ops with same
+        // reference sequence number as this op.
+        // Sysmte ops (when no clients are connected) are the only ops where equation is possible.
+        const diff = message.sequenceNumber - message.minimumSequenceNumber;
+        if (diff < 0 || diff === 0 && message.clientId !== null) {
```

Comment:
Is this situation something we're suspecting of already happening?

Reply:
Not sure what you mean exactly. Nicholas asked me about MSN invariants, and while looking at the code I was not able to find us asserting these invariants, so I'm adding such validation. I do not know if they were ever violated.
That said, check about for clientId === null was clearly one such attempt to assert one invariant, but it was wrong (and was noop).

Reply:
Thanks for clarifying. I was curious whether there was some hypothesis you had in mind.